lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -172,22 +216,41 @@ class IntegrationTestsManager {
         IntegrationArguments.fromBrowser(_browser);
     final int exitCode = await runProcess(
       executable,
-      arguments.getTestArguments(testName, 'profile'),
+      arguments.getTestArguments(testName, mode,
+          isCanvaskitBackend: canvaskitBackend),
       workingDirectory: directory.path,
       environment: enviroment,
     );
 
     if (exitCode != 0) {
+      final String command = arguments.getCommandToRun(testName, mode);
       io.stderr
           .writeln('ERROR: Failed to run test. Exited with exit code $exitCode'
               '. To run $testName locally use the following command:'
-              '\n\n${arguments.getCommandToRun(testName, 'profile')}');
+              '\n\n$command');
       return false;
     } else {
       return true;
     }
   }
 
+  Set<String> _getBuildModes() {
+    Set<String> buildModes;
+    if (_buildModeSelected) {
+      final String mode = IntegrationTestsArgumentParser.instance.buildMode;
+      if (mode == 'debug' && _browser != 'chrome') {
+        throw ToolException('Debug mode is only supported for Chrome.');
+      } else {
+        buildModes = <String>{mode};
+      }
+    } else {
+      buildModes = _browser == 'chrome'
+          ? {'debug', 'profile', 'release'}
+          : {'profile', 'release'};
```

Comment:
I think the built-in knowledge of which browsers should use which modes is adding complexity without much benefit. For example, one might attempt to run a test in Edge, but will need to change `felt` code in order to do it.

I would instead run in one mode only and in the CI script invoke `felt` multiple time with different modes, perhaps even in different subshards.

Reply:
1) I already wrote a design with running multiple shards. Felt can be invoked with multiple modes. Yet the default will be all modes.
2) I think it only requires knowledge of whether that build mode is supported in a browser. Felt still only tests for legacy Edge, therefore it only testing for profile and release mode will be ok. I will leave the tests like this since if a developer runs: `felt test` I want them to see all that is failing and passing instead of running the multiple times with different build modes.